src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -0,0 +1,95 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_config.h"
+
+#if HAVE_LINUX_RTNETLINK_H
```

Comment:
Instead of guarding this entire file out using the preprocessor, I'd prefer that it be conditionally included when building for Linux in `CMakeLists.txt` and that cmake emit an error if `rtnetlink.h` cannot be found.